src/controllers/graphql/user-controller.js

```diff
@@ -23,3 +23,27 @@ export const signup = async (_, { data: user }, { models, secret }) => {
     jwt: token
   }
 }
+
+const getValidatedUserFrom = async (models, userData) => {
+  const validUser = await models.user.findByUsername(userData.username)
+
+  if (!validUser) {
+    throw Error('User not found')
+  }
+
+  if (!validUser.passwordMatches(userData.password)) {
+    throw Error('User not found')
+  }
+
+  return validUser
+}
```

Comment:
What do you think about moving this function to the `User` model with the name `login`, `signin` or `getUserWithCredentials`?